### ISSUE TITLE
Test Kroxylicious against older Kafka Brokers

### ISSUE THREAD

**Issue by k-wall** [kroxylicious/kroxylicious#1111]
Currently the test suite is only exercising the Kroxylicious against the version of Kafka Kroxylicious is build against.  We ought to have a test matrix that tests against earlier versions of Kafka Broker in order to assure compatibility.

We probably need a test matrix and some way of disabling  tests that should only be run for certain version of Kafka.

We might be able to exploit the CONTAINER mode of the Kroxylicious Junit Test Extension to allow the test suite to be executed against older Brokers.

**Comment by showuon** [comment]
We should raise another issue for "testing all versions of Kafka client".

**Comment by k-wall** [comment]
> We should raise another issue for "testing all versions of Kafka client".

https://github.com/kroxylicious/kroxylicious/issues/1131

**Comment by robobario** [comment]
has hopped many milestones without being picked up, removing from milestone. and will remove `triaged` to discuss

**Comment by franvila** [comment]
On system tests we have a constrain here:
* If we run the tests using the `strimzi_test_client` we depend on the version selected from [quay](https://quay.io/repository/strimzi-test-clients/test-clients?tab=tags), and each test client version supports only 2 different minor versions of Kafka (i.e 0.14.0 supports 4.2.0, 4.1.1 and 4.1.0 versions)
* We can use another non JVM test client to test Kroxylicious against older kafka brokers like the [junit-extension](https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/main/.github/workflows/maven.yaml#L92C18-L92C55), such as `python_test_client`

**Comment by k-wall** [comment]
> On system tests we have a constrain here:
> 
> * If we run the tests using the `strimzi_test_client` we depend on the version selected from [quay](https://quay.io/repository/strimzi-test-clients/test-clients?tab=tags), and each test client version supports only 2 different minor versions of Kafka (i.e 0.14.0 supports 4.2.0, 4.1.1 and 4.1.0 versions)
> * We can use another non JVM test client to test Kroxylicious against older kafka brokers like the [junit-extension](https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/main/.github/workflows/maven.yaml#L92C18-L92C55), such as `python_test_client`

This issue is about testing with older brokers, so the client version shouldn't matter.  The client should negotiate down.  
I think your comment belongs on https://github.com/kroxylicious/kroxylicious/issues/1131.

**Comment by franvila** [comment]
> > On system tests we have a constrain here:
> > 
> > * If we run the tests using the `strimzi_test_client` we depend on the version selected from [quay](https://quay.io/repository/strimzi-test-clients/test-clients?tab=tags), and each test client version supports only 2 different minor versions of Kafka (i.e 0.14.0 supports 4.2.0, 4.1.1 and 4.1.0 versions)
> > * We can use another non JVM test client to test Kroxylicious against older kafka brokers like the [junit-extension](https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/main/.github/workflows/maven.yaml#L92C18-L92C55), such as `python_test_client`
> 
> This issue is about testing with older brokers, so the client version shouldn't matter. The client should negotiate down. I think your comment belongs on [#1131](https://github.com/kroxylicious/kroxylicious/issues/1131).

I know it does not matter. I just wanted to highlight that because by default all system tests run using `strimzi_test_client`

**Comment by franvila** [comment]
Another question is whether we want to include this testing on PRs, Main only or both. If we run all tests for each Kafka version it will increase too much the number of jobs running in parallel and we currently have issues with jobs queues on Github.
* For PRs we currently have 1 job per test suite (9 jobs). Adding 5 older Kafka brokers (3.8.0,3.9.0,4.0.0,4.1.0,4.2.0) will multiply this: 45 jobs (for older kafka brokers) + 9 jobs (for current kafka broker).
* For Main we already have 53 jobs running (we include several kafka test clients). If we include only for `python_test_client` we have to sum 45 jobs.

I see this amount of jobs to heavy for our current Github account (only 20 jobs can be run in parallel). What if we select a minimum `regression` tests to be run? We just need to tag them as `REGRESSION`

**Comment by k-wall** [comment]
I agree that it would be an opt in feature.  Activation with a label sounds like a reasonable way to go.
